### ISSUE TITLE
Bug or feat: the whole pipeline relay on the output of the text encoder sdpa undefined behavior

### ISSUE THREAD

**Issue by ValMystletainn** [meituan-longcat/LongCat-Image#70]
currently, longcat image use qwen2.5-VL hidden as text embedding, feeding to dit.

However, I just find we use qwen2.5-VL with **sdpa attention implementation**,  with **padding** to 512/1024 (for t2i, i2i), **with attention mask in vlm** but **no attention mask** in dit.

I just rolling out different prompt embedding, currently (pytorch version 2.5-2.10), and find

1. the embeddings of padding tokens are same in the same prompt, i.e. prompt `<im_start>......., something balala, <pad> <pad> <pad>...` will get the same embedding vectors at the tail.
2. but the embeddings of `<pad>` tokens are different when come across different prompts, i.e. the embedding of padding tokens in  `<im_start>......., prompt_1, <pad> <pad> <pad>...`  are different in `<im_start>......., prompt_2, <pad> <pad> <pad>...`
3. so I dived into the implementation, and find there is attention mask in vlm forward, mask out the padding token, but there is not attention mask in dit, the dit should attention the <pad> token, and actually the pad token significantly affects the outcome quality.

<img width="1378" height="600" alt="Image" src="https://github.com/user-attachments/assets/074dd1a7-f484-4eaa-bf4b-e191e5746e48" />
t2i with tokens pad to 256/512 length.

I think the output with no attention token in sdpa is a undefined behavior, and write this issue to ask is it a design choice. the reasonable design I think is:

1. work flow like flux1 t5 usage. padding to 512 token, full attention in text encoder with pad token and also dit.
2. work flow like qwen-image, padding and mask out the pad token in vlm and also in dit.

I just working on the native resolution training(https://arxiv.org/abs/2307.06304), extending longcat-image edit to multi reference model. So I find the  peculiar and difficult-to-address characteristics that make my trainig process inefficient.
